File: bindings/lua/Polycode/VideoModeChangeInfo.lua
class "VideoModeChangeInfo"


function VideoModeChangeInfo:__getvar(name)
	if name == "xRes" then
		return Polycode.VideoModeChangeInfo_get_xRes(self.__ptr)
	elseif name == "yRes" then
		return Polycode.VideoModeChangeInfo_get_yRes(self.__ptr)
	elseif name == "fullScreen" then
		return Polycode.VideoModeChangeInfo_get_fullScreen(self.__ptr)
	elseif name == "vSync" then
		return Polycode.VideoModeChangeInfo_get_vSync(self.__ptr)
	elseif name == "aaLevel" then
		return Polycode.VideoModeChangeInfo_get_aaLevel(self.__ptr)
	elseif name == "anisotropyLevel" then
		return Polycode.VideoModeChangeInfo_get_anisotropyLevel(self.__ptr)
	elseif name == "retinaSupport" then
		return Polycode.VideoModeChangeInfo_get_retinaSupport(self.__ptr)
	end
end

function VideoModeChangeInfo:__setvar(name,value)
	if name == "xRes" then
		Polycode.VideoModeChangeInfo_set_xRes(self.__ptr, value)
		return true
	elseif name == "yRes" then
		Polycode.VideoModeChangeInfo_set_yRes(self.__ptr, value)
		return true
	elseif name == "fullScreen" then
		Polycode.VideoModeChangeInfo_set_fullScreen(self.__ptr, value)
		return true
	elseif name == "vSync" then
		Polycode.VideoModeChangeInfo_set_vSync(self.__ptr, value)
		return true
	elseif name == "aaLevel" then
		Polycode.VideoModeChangeInfo_set_aaLevel(self.__ptr, value)
		return true
	elseif name == "anisotropyLevel" then
		Polycode.VideoModeChangeInfo_set_anisotropyLevel(self.__ptr, value)
		return true
	elseif name == "retinaSupport" then
		Polycode.VideoModeChangeInfo_set_retinaSupport(self.__ptr, value)
		return true
	end
	return false
end
function VideoModeChangeInfo:__delete()
	if self then Polycode.delete_VideoModeChangeInfo(self.__ptr) end
end

File: bindings/javascript/Polycode/VideoModeChangeInfo.js
function VideoModeChangeInfo() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.VideoModeChangeInfo()
	}
	Object.defineProperties(this, {
		'xRes': { enumerable: true, configurable: true, get: VideoModeChangeInfo.prototype.__get_xRes, set: VideoModeChangeInfo.prototype.__set_xRes},
		'yRes': { enumerable: true, configurable: true, get: VideoModeChangeInfo.prototype.__get_yRes, set: VideoModeChangeInfo.prototype.__set_yRes},
		'fullScreen': { enumerable: true, configurable: true, get: VideoModeChangeInfo.prototype.__get_fullScreen, set: VideoModeChangeInfo.prototype.__set_fullScreen},
		'vSync': { enumerable: true, configurable: true, get: VideoModeChangeInfo.prototype.__get_vSync, set: VideoModeChangeInfo.prototype.__set_vSync},
		'aaLevel': { enumerable: true, configurable: true, get: VideoModeChangeInfo.prototype.__get_aaLevel, set: VideoModeChangeInfo.prototype.__set_aaLevel},
		'anisotropyLevel': { enumerable: true, configurable: true, get: VideoModeChangeInfo.prototype.__get_anisotropyLevel, set: VideoModeChangeInfo.prototype.__set_anisotropyLevel},
		'retinaSupport': { enumerable: true, configurable: true, get: VideoModeChangeInfo.prototype.__get_retinaSupport, set: VideoModeChangeInfo.prototype.__set_retinaSupport}
	})
}

VideoModeChangeInfo.prototype.__get_xRes = function() {
	return Polycode.VideoModeChangeInfo__get_xRes(this.__ptr)
}

VideoModeChangeInfo.prototype.__set_xRes = function(val) {
	Polycode.VideoModeChangeInfo__set_xRes(this.__ptr, val)
}

VideoModeChangeInfo.prototype.__get_yRes = function() {
	return Polycode.VideoModeChangeInfo__get_yRes(this.__ptr)
}

VideoModeChangeInfo.prototype.__set_yRes = function(val) {
	Polycode.VideoModeChangeInfo__set_yRes(this.__ptr, val)
}

VideoModeChangeInfo.prototype.__get_fullScreen = function() {
	return Polycode.VideoModeChangeInfo__get_fullScreen(this.__ptr)
}

VideoModeChangeInfo.prototype.__set_fullScreen = function(val) {
	Polycode.VideoModeChangeInfo__set_fullScreen(this.__ptr, val)
}

VideoModeChangeInfo.prototype.__get_vSync = function() {
	return Polycode.VideoModeChangeInfo__get_vSync(this.__ptr)
}

VideoModeChangeInfo.prototype.__set_vSync = function(val) {
	Polycode.VideoModeChangeInfo__set_vSync(this.__ptr, val)
}

VideoModeChangeInfo.prototype.__get_aaLevel = function() {
	return Polycode.VideoModeChangeInfo__get_aaLevel(this.__ptr)
}

VideoModeChangeInfo.prototype.__set_aaLevel = function(val) {
	Polycode.VideoModeChangeInfo__set_aaLevel(this.__ptr, val)
}

VideoModeChangeInfo.prototype.__get_anisotropyLevel = function() {
	return Polycode.VideoModeChangeInfo__get_anisotropyLevel(this.__ptr)
}

VideoModeChangeInfo.prototype.__set_anisotropyLevel = function(val) {
	Polycode.VideoModeChangeInfo__set_anisotropyLevel(this.__ptr, val)
}

VideoModeChangeInfo.prototype.__get_retinaSupport = function() {
	return Polycode.VideoModeChangeInfo__get_retinaSupport(this.__ptr)
}

VideoModeChangeInfo.prototype.__set_retinaSupport = function(val) {
	Polycode.VideoModeChangeInfo__set_retinaSupport(this.__ptr, val)
}

Duktape.fin(VideoModeChangeInfo.prototype, function (x) {
	if (x === VideoModeChangeInfo.prototype) {
		return;
	}
	Polycode.VideoModeChangeInfo__delete(x.__ptr)
})
